### DISCUSSION TITLE
Environments only available in public or enterprise

### DISCUSSION THREAD

**Discussion by chrisdrobison** [community/community#9150] [Enterprise]
I guess I would love to understand the reasoning behind only having environments in public and enterprise. I've been messing around with transferring some private deployments from Azure DevOps where environments are just included to GitHub Actions and this is essentially a show stopper for me. There's no reason to bring anything over now in my opinion. This seems like an odd exclusion. Would it be possible to reconsider this?

**Comment by ethomson** [comment]
Hi @chrisdrobison - I appreciate the feedback.  We don't have any immediate plans to change this, but I'll pass your feedback along to the team.

**Comment by galah92** [comment]
I'd like to join to the request. This is very critical for my organization as well, and I'm still trying to bend the way my workflows are structured in order to correctly support deployments to multiple environments. Highly critical for me.

**Comment by cemo** [comment]
I can not understand why this feature is in enterprise plan. It is definitely not an enterprise feature.

**Comment by IanMoroney** [comment]
I currently operate on the free plan, and I'd consider `paying` for the `Teams` plan immediately if the `Environments` feature was included in it.
Please reconsider which plans this feature is available for, considering the fact that other compeditors tools (octopus, teamcity, jenkins, circleci) all allow for this out of the box.
Additionally, Azure Devops allows for this, as was already pointed out.

For most SMEs, the `Teams` plan is perfectly suited to their needs, and should really promote good `Devops practices`, which is why this feature should be included in the Teams plan.

Not everyone needs all the features that Enterprise offers, nor can everyone afford it.

**Comment by DamienDennehy** [comment]
It's a tough sell to financial stakeholders. "If we want Environments we have to quadruple our GitHub costs".

**Comment by jmbenedetto** [comment]
Hi! I have only public repos, but do not see the "Environment" option at the organization level. Would you know it it should be available?

**Comment by chrisdrobison** [comment]
I'm seeing now that environments are available pretty much everywhere, but environment protection rules are not. Again, this is not an enterprise feature, this should just be something intrinsic to environments everywhere. For my small business project, I have no way doing deploys in GitHub Actions because I can't gate environments. I understand all the SAML, SSO and other things being enterprise, but protection rules should be everywhere.
